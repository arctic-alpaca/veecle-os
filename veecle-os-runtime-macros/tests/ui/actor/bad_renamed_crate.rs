#[derive(Debug, PartialEq, Clone, Default)]
pub struct Sensor(pub u8);

#[veecle_os_runtime_macros::actor(crate = "::veecle_os_runtime")]
async fn macro_test_actor1() -> std::convert::Infallible {
    unreachable!("We only care about the code compiling.");
}

#[veecle_os_runtime_macros::actor(crate())]
async fn macro_test_actor2() -> std::convert::Infallible {
    unreachable!("We only care about the code compiling.");
}

#[veecle_os_runtime_macros::actor(crate = ::veecle_os_runtime, crate = ::veecle_os_runtime2)]
async fn macro_test_actor3() -> std::convert::Infallible {
    unreachable!("We only care about the code compiling.");
}

fn main() {
}
