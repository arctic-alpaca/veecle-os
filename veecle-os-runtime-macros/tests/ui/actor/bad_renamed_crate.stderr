error: invalid value for `crate`, must be a simple path
 --> tests/ui/actor/bad_renamed_crate.rs:4:43
  |
4 | #[veecle_os_runtime_macros::actor(crate = "::veecle_os_runtime")]
  |                                           ^^^^^^^^^^^^^^^^^^^^^

error: `crate` must be a name value pair (`crate = veecle_os_runtime`)
 --> tests/ui/actor/bad_renamed_crate.rs:9:35
  |
9 | #[veecle_os_runtime_macros::actor(crate())]
  |                                   ^^^^^^^

error: setting `crate` argument multiple times
  --> tests/ui/actor/bad_renamed_crate.rs:14:64
   |
14 | #[veecle_os_runtime_macros::actor(crate = ::veecle_os_runtime, crate = ::veecle_os_runtime2)]
   |                                                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
